src/Core/Grid/DataProvider/GridDataProviderInterface.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\DataProvider;
+
+/**
+ * Interface GridDataProviderInterface defines contract for grid data providers
+ */
+interface GridDataProviderInterface
+{
+    /**
+     * Get filtered & paginated rows from any data source (database, API or any other)
+     *
+     * @param array $filters
+     *
+     * @return array
```

Comment:
I'd like to see a real RowsCollection here, using an object is easier for validation.

If one property is missing on your list, or if we want to allow a specific behavior for one column, it's better to manage it in his own object. (It wasn't finished, but you can spy ItemCollection/ItemInterface/Item from my PR)

Reply:
``getRows()`` returns rows data from database (in most cases), are you sure collection in necessary here?

Reply:
Yes, if we need to do operations on Rows, I prefer to keep these operations in only one place: RowsCollection.

Reply:
Note that I'm totaly open to an abstract class "Collection" and make every collection of PrestaShop extends it to make this work "less annoying".